// vuepress-theme-hope@2.0.0-rc.87/lib/client/components/info/WordInfo.js
import { defineComponent, h } from "vue";
import { WordIcon } from "@theme-hope/components/info/icons";
import { useMetaLocale } from "@theme-hope/composables/info/useMetaLocale";
import { usePure } from "@theme-hope/composables/usePure";
export default defineComponent({
    name: "ReadTimeInfo",
    inheritAttrs: false,
    props: {
        /**
         * Reading time information
         *
         * é˜…è¯»æ—¶é—´ä¿¡æ¯
         */
        readingTime: Object,
        /**
         * Reading time locale
         *
         * é˜…è¯»æ—¶é—´è¯­è¨€çŽ¯å¢ƒ
         */
        readingTimeLocale: Object,
    },
    setup(props) {
        const metaLocale = useMetaLocale();
        const isPure = usePure();
        return () => props.readingTimeLocale?.words
            ? h("span", {
                class: "page-word-info",
                "aria-label": `${metaLocale.value.words}${isPure.value ? "" : "ðŸ” "}`,
                ...(isPure.value ? {} : { "data-balloon-pos": "up" }),
            }, [
                h(WordIcon),
                h("span", props.readingTimeLocale.words),
                h("meta", {
                    property: "wordCount",
                    content: props.readingTime?.words,
                }),
            ])
            : null;
    },
});
